File: Master/1st Course/Fall semester/Paralelie procesi un valoda ADA/Lab8/Task2/C#/MatrixThread.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;

namespace ThreadingTask {

    class MatrixThread {
        // Private class members.
        private Thread thread;
        private int[,] matrix;
        private int startRow;
        MatrFun matrFun;

        public MatrixThread(
            int[,] matr,
            string name,
            int row,
            ThreadPriority priority,
            MatrFun objMatr) {

                matrix = matr;
                startRow = row;
                matrFun = objMatr;

                thread = new Thread(this.Run);
                thread.Name = name;
                thread.Priority = priority;
                thread.Start();
        }

        private void Run() {
            // Synchronizing or locking object.
            for (int i = startRow; i < matrix.GetLength(0); i += 2) {
                lock (this.matrFun) {
                    matrFun.PrintRow(i, matrix);
                }
            }
        }
    }
}

File: Master/1st Course/Fall semester/Paralelie procesi un valoda ADA/Lab8/Task2/Java/MatrixThread.java
package lab8.threading;

public class MatrixThread implements Runnable {
    private Thread thread;
    private int[][] matrix;
    private int startRow;
    private MatrFun matrFun;

    public MatrixThread(
            int[][] matr,
            String name,
            int row,
            int priority,
            MatrFun objMatr) { 
        
        matrix = matr;
        startRow = row;
        matrFun = objMatr;
        
        thread = new Thread(this);
        thread.setName(name);
        thread.setPriority(priority);
        thread.start();
    }
    public void run() {
        for (int i = startRow; i < matrix.length; i += 2) {
            synchronized(this.matrFun) {
                matrFun.printRow(i, matrix);
            }
        }
    }
}
